app/models/spree/order.rb

```diff
@@ -621,6 +621,15 @@ def after_delivery_state?
       state.in?(["payment", "confirmation"])
     end
 
+    # @return [BigDecimal] The rate of the voucher if applied to the order
+    def applied_voucher_rate
+      # As an order can have only one voucher, hence using +take+ because each voucher adjustment will have the same voucher
+      return BigDecimal(0) unless (voucher_adjustment = voucher_adjustments.take)
+
+      voucher = voucher_adjustment.originator
```

Comment:
I wonder if this can be better expressed as AR association. `has_many :vouchers, through: :voucher_adjustments` should work. Maybe even `has_one`?

Reply:
Yes, this is the best case scenario for the current situation at hand.
I believe when vouchers were introduced, we had it in mind that in the near future we might be needing more than 1 vouchers applied to an order.

Hey @RachL - It would be great if we have some kind of confirmation on whether we are planning to introduce multiple vouchers to apply to an order. 
If it's not planned in the near future, then I believe we should create a good-first-issue issue and incorporate this comment. This would help us clean up some logic in code. Thanks.

Reply:
Yes it's the best scenario for the situation at hand.
I'd go for the has_many because this would help us incorporate multiple vouchers if in future we have this feature without making any schema level changes.

We can have an issue to incorporate this comment and use this relationship wherever required. What do you say?

Reply:
I think it's fine to do it in this issue. I don't think we had a need for it so far so that's probably why it hasn't been done, and yes we should use `has_many`.